gazebosim/gz-sim
# Fix making breadcrumb static if it's a nested model

# 🦟 Bug fix

Fixes https://github.com/gazebosim/gazebo_test_cases/issues/1593

Breadcrumbs are made static  some time after being deployed. This is done by attaching a fixed joint between its canonical link and a dummy static model. However, the logic for finding the canonical link did not work if the breadcrumb is a nested model. This PR fixes finding the canonical link in a nested breadcrumb model so that it can be made static.

To test

launch breadcrumbs.sdf world

```
gz sim -v 4 -r breadcrumbs.sdf
```

Deploy B2

```
gz topic -t "/B2/deploy" -m gz.msgs.Empty -p "unused: true"
```

Wait a few seconds and make sure no error msgs such as `[error] Failed to make breadcrumb 'B2_0' static.` is printed.

The deployed breadcrumb model should be static.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
